src/operator/l2_normalization-inl.h

```diff
@@ -108,8 +108,8 @@ class L2NormalizationOp : public Operator {
       Tensor<xpu, 2> norm = out_data[l2_normalization::kNorm]
         .get_with_shape<xpu, 2, real_t>(norm_shape, s);
       norm = reduce_with_axis<red::sum, false>(F<mxnet::op::mshadow_op::square>(data), 2);
-      norm = F<mxnet::op::mshadow_op::square_root>(norm);
-      out = data / broadcast_with_axis(norm + param_.eps, 1, dshape[2]);
+      norm = F<mxnet::op::mshadow_op::square_root>(norm + param_.eps);
```

Comment:
We notice that test_l2_normalization() randomly fails for the backward numerical gradient check. We assume this will solve the problem: adding the eps inside the norm makes the backward gradient more consistent with its numerial counterpart.